### PR TITLE
Add PreferNativeArm64 property

### PR DESCRIPTION
**Pull request by haileymck**
For [Task](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2060489)

This new property only applies to project that target .NET framework ad have an executable output. 

This is what the UI looks like, let me know if there is a better fitting description.
image

image

 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/project-system/pull/9507)

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/AbstractProjectConfigurationProperties.cs [hunk 1]
```diff
@@ -218,5 +218,6 @@ public string DebugInfo
         public bool CodeAnalysisIgnoreBuiltInRules { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
         public bool CodeAnalysisFailOnMissingRules { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
         public bool Prefer32Bit { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public bool PreferNativeArm64 { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/AbstractProjectConfigurationProperties.cs:221]: You don't need to add this. This class serves as the base implementation of the `ProjectConfigurationProperties3` interface; it doesn't need any functionality that isn't exposed by that interface.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Shipped.txt [hunk 2]
```diff
@@ -217,6 +217,8 @@ Microsoft.VisualStudio.ProjectSystem.VS.Properties.AbstractProjectConfigurationP
 Microsoft.VisualStudio.ProjectSystem.VS.Properties.AbstractProjectConfigurationProperties.PlatformTarget.set -> void
 Microsoft.VisualStudio.ProjectSystem.VS.Properties.AbstractProjectConfigurationProperties.Prefer32Bit.get -> bool
 Microsoft.VisualStudio.ProjectSystem.VS.Properties.AbstractProjectConfigurationProperties.Prefer32Bit.set -> void
+Microsoft.VisualStudio.ProjectSystem.VS.Properties.AbstractProjectConfigurationProperties.PreferNativeArm64.get -> bool
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Shipped.txt:220]: And if we don't need a new public member on `AbstractProjectConfigurationProperties` we don't need these lines, either.

**Review by tmeschter — CHANGES_REQUESTED** [review]

**Comment by tmeschter** [comment]
Another thing to look into is whether checking "Prefer Native ARM64" in a non-SDK-style C# project makes configuration-specific changes (e.g. with the value set in one or more configuration-specific `PropertyGroup`s, as shown in your screenshot) or not. We should be consistent with that, whether or not that matches the behavior of "Prefer 32-bit".

**Comment by haileymck** [comment]
> Another thing to look into is whether checking "Prefer Native ARM64" in a non-SDK-style C# project makes configuration-specific changes (e.g. with the value set in one or more configuration-specific `PropertyGroup`s, as shown in your screenshot) or not. We should be consistent with that, whether or not that matches the behavior of "Prefer 32-bit".

yes, this is the behavior for the "Prefer32Bit" and other properties: 
image

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralConfiguredBrowseObject.xaml
Comment (haileymck) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralConfiguredBrowseObject.xaml:1]: Should I remove this as well? I think yes...

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/AbstractProjectConfigurationProperties.cs (hunk 1 above)
Comment (haileymck) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/AbstractProjectConfigurationProperties.cs]: removed!

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Shipped.txt (hunk 2 above)
Comment (haileymck) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/PublicAPI.Shipped.txt]: removed!

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralConfiguredBrowseObject.xaml
Comment (tmeschter) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralConfiguredBrowseObject.xaml:1]: Yeah, you could remove it. It shouldn't be needed to support the UI.

**Review by drewnoakes — APPROVED** [review]
LGTM

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralConfiguredBrowseObject.xaml
Comment (haileymck) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/GeneralConfiguredBrowseObject.xaml:1]: removed!

**Review by drewnoakes — APPROVED** [review]

**Review by tmeschter — APPROVED** [review]

**Pull request merged**